3_computer_vision/ml_project_4_face_detection/backend/service_layer.py

```diff
@@ -0,0 +1,24 @@
+from PIL import Image
+from image_utils import (
+    draw_boxes, 
+    get_gcp_bounding_boxes,
+    print_face_diagnostics
+)
+import io
+
+async def detect_faces(contents: io.BytesIO, client) -> io.BytesIO:
```

Comment:
What about face encoding or recognition? You can just call `VGGFace().predict(img)` it returns recognition results with top matches.

Reply:
I will add that as well.